nixos/modules/services/x11/xserver.nix

```diff
@@ -651,25 +651,33 @@ in
     system.extraDependencies = singleton (pkgs.runCommand "xkb-layouts-exist" {
       inherit (cfg) layout xkbDir;
     } ''
-      if sed -n -e ':i /^! \(layout\|variant\) *$/ {
-          :l; n; /^!/bi; s/^ *\([^ ]\+\).*/\1/p; tl
-         }' "$xkbDir/rules/base.lst" | grep -qxF "$layout"
-      then
-        touch "$out"
+      IFS=, read -a layouts <<<'${cfg.layout}'
+
+      missing=()
+      for layout in "''${layouts[@]}" ; do
+        if ! sed -n -e ':i /^! \(layout\|variant\) *$/ {
+            :l; n; /^!/bi; s/^ *\([^ ]\+\).*/\1/p; tl
+           }' "$xkbDir/rules/base.lst" | grep -qxF "$layout"
```

Comment:
That would result in it searching the file named by `"$layout"`. But in a similar vein, I should add `-e`.